RecoVertex/PrimaryVertexProducer/interface/VertexTimeAlgorithmLegacy4D.h

```diff
@@ -0,0 +1,23 @@
+#ifndef usercode_PrimaryVertexAnalyzer_VertexTimeLegacy4D_h
+#define usercode_PrimaryVertexAnalyzer_VertexTimeLegacy4D_h
+
+#include "VertexTimeAlgorithmBase.h"
+
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+
+class VertexTimeAlgorithmLegacy4D : public VertexTimeAlgorithmBase {
+public:
+  VertexTimeAlgorithmLegacy4D(const edm::ParameterSet& conf, edm::ConsumesCollector& iC);
+  ~VertexTimeAlgorithmLegacy4D() override = default;
+
+  static void fillPSetDescription(edm::ParameterSetDescription& iDesc);
+
+  void setEvent(edm::Event& iEvent, edm::EventSetup const& iSetup) override;
+
+  bool vertexTime(float& vtxTime, float& vtxTimeError, TransientVertex const& vtx) const override;
+
+  static constexpr float defaultInvalidTrackReso = 0.350f;
```

Comment:
this is a temporary solution, it would make sense to move the original definition

https://github.com/cms-sw/cmssw/blob/master/TrackingTools/TransientTrack/src/TransientTrackBuilder.cc#L17 

into tha class' header, and use that header here as well. To be done in a next commit